hub/policy/src/main/java/uk/gov/ida/hub/policy/domain/controller/AbstractMatchRequestSentStateController.java

```diff
@@ -80,55 +79,83 @@ public void handleRequestFailure() {
         stateTransitionAction.transitionTo(matchingServiceRequestErrorState);
     }
 
-    protected U getSuccessfulMatchState(MatchFromMatchingService responseFromMatchingService) {
-        String matchingServiceAssertion = responseFromMatchingService.getMatchingServiceAssertion();
-        validator.validate(responseFromMatchingService.getLevelOfAssurance(), state.getIdpLevelOfAssurance());
-        String requestIssuerId = state.getRequestIssuerEntityId();
-        return createSuccessfulMatchState(matchingServiceAssertion, requestIssuerId);
+    @Override
+    public ResponseProcessingDetails getResponseProcessingDetails() {
+        if (matchingServiceRequestHasTimedOut()) {
+            LOG.error("Matching service request timed out for session {}", state.getSessionId());
+            MatchingServiceRequestErrorState matchingServiceRequestErrorState = new MatchingServiceRequestErrorState(
+                    state.getRequestId(),
+                    state.getRequestIssuerEntityId(),
+                    state.getSessionExpiryTimestamp(),
+                    state.getAssertionConsumerServiceUri(),
+                    state.getIdentityProviderEntityId(),
+                    state.getRelayState(),
+                    state.getSessionId(),
+                    state.getTransactionSupportsEidas());
+
+            stateTransitionAction.transitionTo(matchingServiceRequestErrorState);
+            return createResponseProcessingDetailsForStatus(ResponseProcessingStatus.SHOW_MATCHING_ERROR_PAGE);
+        }
+
+        return createResponseProcessingDetailsForStatus(ResponseProcessingStatus.WAIT);
+    }
+
+    @Override
+    public ResponseFromHub getErrorResponse() {
+        return responseFromHubFactory.createNoAuthnContextResponseFromHub(
+                state.getRequestId(),
+                state.getRelayState(),
+                state.getRequestIssuerEntityId(),
+                state.getAssertionConsumerServiceUri()
+        );
     }
 
     @Override
     public final void handleMatchResponseFromMatchingService(MatchFromMatchingService responseFromMatchingService) {
         validateResponse(responseFromMatchingService);
-
-        State nextState = getNextStateForMatch(responseFromMatchingService);
-
-        stateTransitionAction.transitionTo(nextState);
+        levelOfAssuranceValidator.validate(responseFromMatchingService.getLevelOfAssurance(), state.getIdpLevelOfAssurance());
```

Comment:
So we added the extra validation for consistency sake?

Reply:
Basically yes. I just moved the repeated logic from the subclasses to the top level class as it was repeated. So I wanted to tidy up a bit without changing the behaviour. Do you think this check is unecessary?